Fix stereo issue with different SDR devices

tomneda opened the pull request:
Hi Jan,
as mentioned in private emails to you longer ago I still had a stereo issue with my new  SDRPlay RSPdx device that it sounds only like "mono" whereas the DabStick works quite well.

I go on checking this further and I have this pull request  for you where it at least these two devices work well, but the fix sounds reasonable so I expect that other devices will working well, too.

The main change I did was the pilotDelay where the calculation is now regarding the pilot filter pulse center not the whole filter length at all:
- new: `pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE / 2 - 1) * OMEGA_PILOT;`
- old:    `pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE) * OMEGA_PILOT;`

What is not so clear to me, I had also change the mixing phase about 90 degree (at the doubled 38 kHz side) and so I used the sin() instead of the cos() function to do the LRDiff mix.

What could be still a (new) issue: I am not sure if I should shift about +90 or -90 degree. So it could be a minus in front of the sin necessary (or better/faster, shift +/-45 degree (we double later!)  further in pilotDelay. With this issue we would have a swapped L/R channel. Maybe you have a reliable test signal for this?

---
If you want to test my fix with other devices you will find a test branch in my repo:
https://github.com/tomneda/sdr-j-fm/tree/develop/stereo_issue_test

Doing this for a test:
1) Select "S|S" to hear only the sideband signal.
2) Check different stereo broadcast stations as some could have pilot frequency inaccuracy.
3) Check each station with the IQ balance slider whether the signal get minimal at center "0" position

The IQ balance slider adds an artificial phase shift (in degree) to the pilot tone

---

BTW: I wish you this way a happy X-mas and so many health and luck what you can get :-) 

BR 
Thomas

JvanKatwijk commented:
Interesting, but strange

The mod seems to work well, I'll test tomorrow with some other devices.
The  strange thing is that the delay of the pilotBandfilter is the size of
the fft - the size of the fir filtersize that is used,
the overlap size is here the same as the filtersize, so before spitting out
a filtered sample, a buffer with the size (fftsize - filtersize_
is filled.

One more comment:
A radio friend/colegue was not very impressed by the band filtering, he is
a DX-er. So I modified the fit band filter with
an FFT filter with a huge size, filtering is of course much better, but cpu
use increases when the filter is used, so
I changed it such that if the filter id "OFF" it is not in the processing
chain


Anyway,
I wish you fine Xmas and a happy - and healthy -  2023

best
jan


Op vr 23 dec. 2022 om 18:12 schreef tomneda ***@***.***>:

> Hi Jan,
> as mentioned in private emails to you longer ago I still had a stereo
> issue with my new SDRPlay RSPdx device that it sounds only like "mono"
> whereas the DabStick works quite well.
>
> I go on checking this further and I have this pull request for you where
> it at least these two devices work well, but the fix sounds reasonable so I
> expect that other devices will working well, too.
>
> The main change I did was the pilotDelay where the calculation is now
> regarding the pilot filter pulse center not the whole filter length at all:
>
>    - new: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE / 2 - 1) *
>    OMEGA_PILOT;
>    - old: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE) * OMEGA_PILOT;
>
> What is not so clear to me, I had also change the mixing phase about 90
> degree (at the doubled 38 kHz side) and so I used the sin() instead of the
> cos() function to do the LRDiff mix.
>
> What could be still a (new) issue: I am not sure if I should shift about
> +90 or -90 degree. So it could be a minus in front of the sin necessary (or
> better/faster, shift +/-45 degree (we double later!) further in pilotDelay.
> With this issue we would have a swapped L/R channel. Maybe you have a
> reliable test signal for this?
> ------------------------------
>
> If you want to test my fix with other devices you will find a test branch
> in my repo:
> https://github.com/tomneda/sdr-j-fm/tree/develop/stereo_issue_test
>
> Doing this for a test:
>
>    1. Select "S|S" to hear only the sideband signal.
>    2. Check different stereo broadcast stations as some could have pilot
>    frequency inaccuracy.
>    3. Check each station with the IQ balance slider whether the signal
>    get minimal at center "0" position
>
> The IQ balance slider adds an artificial phase shift (in degree) to the
> pilot tone
> ------------------------------
>
> BTW: I wish you this way a happy X-mas and so many health and luck what
> you can get :-)
>
> BR
> Thomas
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/JvanKatwijk/sdr-j-fm/pull/10
> Commit Summary
>
>    - fee74a2
>    <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/commits/fee74a2f540900ddb292a1e36c2d97b27d426eda>
>    Fix stereo issue with different SDR devices
>
> File Changes
>
> (2 files <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files>)
>
>    - *M* .gitignore
>    <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (1)
>    - *M* src/fm/fm-processor.cpp
>    <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-e8d084e4e77aa8b8e699b0d278fedac65a272893601b67dfa5caf0d806cfb521>
>    (4)
>
> Patch Links:
>
>    - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.patch
>    - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQDM5DSGI2HLAGSS5KTWOXMOZANCNFSM6AAAAAATH5TMZ4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Jan van Katwijk

tomneda commented:
Hello Jan,
thanks that you like the mod.

Yes you are right with the delay if you count only the loaded samples into the pilot FFT filter, but the filtering itself takes the half time of the underlying impulse response of the Sinc function? So I guess the substracted filtersize has to be halfed. The additional minus 1 I got by experiment and has a big effect if it is there or not.

I already saw the new steep IF filtering. Your DX friend needs that really for Wideband-FM or tries he (my experimental intended) AM mode where the 15 or 9 kHz are sensful and should be deeper?

Maybe you make the filtersize dependend on the needed bandwidth, e.g. using a big filtersize <20kHz and a smaller one above?

Thanks for the x-max greetings I can only repeat this again.

PS: meanwhile I sent you a further pull request, hope they are ok for you.

BR
Thomas 

23.12.2022 20:43:06 JvanKatwijk ***@***.***>:

> Interesting, but strange
> 
> The mod seems to work well, I'll test tomorrow with some other devices.
> The strange thing is that the delay of the pilotBandfilter is the size of
> the fft - the size of the fir filtersize that is used,
> the overlap size is here the same as the filtersize, so before spitting out
> a filtered sample, a buffer with the size (fftsize - filtersize_
> is filled.
> 
> One more comment:
> A radio friend/colegue was not very impressed by the band filtering, he is
> a DX-er. So I modified the fit band filter with
> an FFT filter with a huge size, filtering is of course much better, but cpu
> use increases when the filter is used, so
> I changed it such that if the filter id "OFF" it is not in the processing
> chain
> 
> 
> Anyway,
> I wish you fine Xmas and a happy - and healthy - 2023
> 
> best
> jan
> 
> 
> Op vr 23 dec. 2022 om 18:12 schreef tomneda ***@***.***>:
> 
>> Hi Jan,
>> as mentioned in private emails to you longer ago I still had a stereo
>> issue with my new SDRPlay RSPdx device that it sounds only like "mono"
>> whereas the DabStick works quite well.
>>
>> I go on checking this further and I have this pull request for you where
>> it at least these two devices work well, but the fix sounds reasonable so I
>> expect that other devices will working well, too.
>>
>> The main change I did was the pilotDelay where the calculation is now
>> regarding the pilot filter pulse center not the whole filter length at all:
>>
>> - new: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE / 2 - 1) *
>> OMEGA_PILOT;
>> - old: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE) * OMEGA_PILOT;
>>
>> What is not so clear to me, I had also change the mixing phase about 90
>> degree (at the doubled 38 kHz side) and so I used the sin() instead of the
>> cos() function to do the LRDiff mix.
>>
>> What could be still a (new) issue: I am not sure if I should shift about
>> +90 or -90 degree. So it could be a minus in front of the sin necessary (or
>> better/faster, shift +/-45 degree (we double later!) further in pilotDelay.
>> With this issue we would have a swapped L/R channel. Maybe you have a
>> reliable test signal for this?
>> ------------------------------
>>
>> If you want to test my fix with other devices you will find a test branch
>> in my repo:
>> https://github.com/tomneda/sdr-j-fm/tree/develop/stereo_issue_test
>>
>> Doing this for a test:
>>
>> 1. Select "S|S" to hear only the sideband signal.
>> 2. Check different stereo broadcast stations as some could have pilot
>> frequency inaccuracy.
>> 3. Check each station with the IQ balance slider whether the signal
>> get minimal at center "0" position
>>
>> The IQ balance slider adds an artificial phase shift (in degree) to the
>> pilot tone
>> ------------------------------
>>
>> BTW: I wish you this way a happy X-mas and so many health and luck what
>> you can get :-)
>>
>> BR
>> Thomas
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>> https://github.com/JvanKatwijk/sdr-j-fm/pull/10
>> Commit Summary
>>
>> - fee74a2
>> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/commits/fee74a2f540900ddb292a1e36c2d97b27d426eda>
>> Fix stereo issue with different SDR devices
>>
>> File Changes
>>
>> (2 files <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files>)
>>
>> - *M* .gitignore
>> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>> (1)
>> - *M* src/fm/fm-processor.cpp
>> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-e8d084e4e77aa8b8e699b0d278fedac65a272893601b67dfa5caf0d806cfb521>
>> (4)
>>
>> Patch Links:
>>
>> - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.patch
>> - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.diff
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10>, or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQDM5DSGI2HLAGSS5KTWOXMOZANCNFSM6AAAAAATH5TMZ4>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
> 
> 
> -- 
> Jan van Katwijk
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/JvanKatwijk/sdr-j-fm/pull/10#issuecomment-1364271809], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AYYTND5CX3O3KT6BCUZE7IDWOX6ERANCNFSM6AAAAAATH5TMZ4].
> You are receiving this because you authored the thread.[Verfolgungsbild][https://github.com/notifications/beacon/AYYTND2CFV2EQ3VNSCFLMJ3WOX6ERA5CNFSM6AAAAAATH5TMZ6WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSRKERMC.gif]
>

JvanKatwijk commented:
well,
I liiked into the code and saw something strange that explains why the
delay was halved
The delay is used to copmpute the right phase of the signal used to compute
the L-R signal and the rds,
just by subtracting delay times the phase difference.
On line 791 for whatever reason the phase was computed based on the double
delay

No idea where the "2 *" came from, it is removed now (and the delay not
halved

best
jan


Op za 24 dec. 2022 om 01:36 schreef tomneda ***@***.***>:

> Hello Jan,
> thanks that you like the mod.
>
> Yes you are right with the delay if you count only the loaded samples into
> the pilot FFT filter, but the filtering itself takes the half time of the
> underlying impulse response of the Sinc function? So I guess the
> substracted filtersize has to be halfed. The additional minus 1 I got by
> experiment and has a big effect if it is there or not.
>
> I already saw the new steep IF filtering. Your DX friend needs that really
> for Wideband-FM or tries he (my experimental intended) AM mode where the 15
> or 9 kHz are sensful and should be deeper?
>
> Maybe you make the filtersize dependend on the needed bandwidth, e.g.
> using a big filtersize <20kHz and a smaller one above?
>
> Thanks for the x-max greetings I can only repeat this again.
>
> PS: meanwhile I sent you a further pull request, hope they are ok for you.
>
> BR
> Thomas
>
> 23.12.2022 20:43:06 JvanKatwijk ***@***.***>:
>
> > Interesting, but strange
> >
> > The mod seems to work well, I'll test tomorrow with some other devices.
> > The strange thing is that the delay of the pilotBandfilter is the size of
> > the fft - the size of the fir filtersize that is used,
> > the overlap size is here the same as the filtersize, so before spitting
> out
> > a filtered sample, a buffer with the size (fftsize - filtersize_
> > is filled.
> >
> > One more comment:
> > A radio friend/colegue was not very impressed by the band filtering, he
> is
> > a DX-er. So I modified the fit band filter with
> > an FFT filter with a huge size, filtering is of course much better, but
> cpu
> > use increases when the filter is used, so
> > I changed it such that if the filter id "OFF" it is not in the processing
> > chain
> >
> >
> > Anyway,
> > I wish you fine Xmas and a happy - and healthy - 2023
> >
> > best
> > jan
> >
> >
> > Op vr 23 dec. 2022 om 18:12 schreef tomneda ***@***.***>:
> >
> >> Hi Jan,
> >> as mentioned in private emails to you longer ago I still had a stereo
> >> issue with my new SDRPlay RSPdx device that it sounds only like "mono"
> >> whereas the DabStick works quite well.
> >>
> >> I go on checking this further and I have this pull request for you where
> >> it at least these two devices work well, but the fix sounds reasonable
> so I
> >> expect that other devices will working well, too.
> >>
> >> The main change I did was the pilotDelay where the calculation is now
> >> regarding the pilot filter pulse center not the whole filter length at
> all:
> >>
> >> - new: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE / 2 - 1) *
> >> OMEGA_PILOT;
> >> - old: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE) * OMEGA_PILOT;
> >>
> >> What is not so clear to me, I had also change the mixing phase about 90
> >> degree (at the doubled 38 kHz side) and so I used the sin() instead of
> the
> >> cos() function to do the LRDiff mix.
> >>
> >> What could be still a (new) issue: I am not sure if I should shift about
> >> +90 or -90 degree. So it could be a minus in front of the sin necessary
> (or
> >> better/faster, shift +/-45 degree (we double later!) further in
> pilotDelay.
> >> With this issue we would have a swapped L/R channel. Maybe you have a
> >> reliable test signal for this?
> >> ------------------------------
> >>
> >> If you want to test my fix with other devices you will find a test
> branch
> >> in my repo:
> >> https://github.com/tomneda/sdr-j-fm/tree/develop/stereo_issue_test
> >>
> >> Doing this for a test:
> >>
> >> 1. Select "S|S" to hear only the sideband signal.
> >> 2. Check different stereo broadcast stations as some could have pilot
> >> frequency inaccuracy.
> >> 3. Check each station with the IQ balance slider whether the signal
> >> get minimal at center "0" position
> >>
> >> The IQ balance slider adds an artificial phase shift (in degree) to the
> >> pilot tone
> >> ------------------------------
> >>
> >> BTW: I wish you this way a happy X-mas and so many health and luck what
> >> you can get :-)
> >>
> >> BR
> >> Thomas
> >> ------------------------------
> >> You can view, comment on, or merge this pull request online at:
> >>
> >> https://github.com/JvanKatwijk/sdr-j-fm/pull/10
> >> Commit Summary
> >>
> >> - fee74a2
> >> <
> https://github.com/JvanKatwijk/sdr-j-fm/pull/10/commits/fee74a2f540900ddb292a1e36c2d97b27d426eda
> >
> >> Fix stereo issue with different SDR devices
> >>
> >> File Changes
> >>
> >> (2 files <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files>)
> >>
> >> - *M* .gitignore
> >> <
> https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947
> >
> >> (1)
> >> - *M* src/fm/fm-processor.cpp
> >> <
> https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-e8d084e4e77aa8b8e699b0d278fedac65a272893601b67dfa5caf0d806cfb521
> >
> >> (4)
> >>
> >> Patch Links:
> >>
> >> - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.patch
> >> - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.diff
> >>
> >> —
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10>, or unsubscribe
> >> <
> https://github.com/notifications/unsubscribe-auth/ACCPHQDM5DSGI2HLAGSS5KTWOXMOZANCNFSM6AAAAAATH5TMZ4
> >
> >> .
> >> You are receiving this because you are subscribed to this thread.Message
> >> ID: ***@***.***>
> >>
> >
> >
> > --
> > Jan van Katwijk
> >
> > —
> > Reply to this email directly, view it on GitHub[
> https://github.com/JvanKatwijk/sdr-j-fm/pull/10#issuecomment-1364271809],
> or unsubscribe[
> https://github.com/notifications/unsubscribe-auth/AYYTND5CX3O3KT6BCUZE7IDWOX6ERANCNFSM6AAAAAATH5TMZ4
> ].
> > You are receiving this because you authored the thread.[Verfolgungsbild][
> https://github.com/notifications/beacon/AYYTND2CFV2EQ3VNSCFLMJ3WOX6ERA5CNFSM6AAAAAATH5TMZ6WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSRKERMC.gif
> ]
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10#issuecomment-1364422930>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQED7WEEETO5WQT3QA3WOZAPNANCNFSM6AAAAAATH5TMZ4>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>


-- 
Jan van Katwijk

tomneda commented:
Hi Jan,

I fear this is not correct :)

Or have you really test this valid, e.g. with my mentioned test branch?

The factor 2 is simple the fact that you need 38 kHz for the LR mix, so you need a double phase step each sample. Formerly there was a further line with factor 3 for the 57 kHz RDS mix signal, too.

The pilotDelay is calculated on the 19kHz pilot so that factor of 2 has to be applied there, too.

So, for me, my version before was the (more) correct one.

I am only not sure if the LRDiff  should be inverted because the audio of L/R could be swapped. Maybe you have an Stereo test generator at your side? Or a reference WAV file which would show the correct channel assignment? But I can do that test later also.

BR
Thomas

24.12.2022 10:14:26 JvanKatwijk ***@***.***>:

> well,
> I liiked into the code and saw something strange that explains why the
> delay was halved
> The delay is used to copmpute the right phase of the signal used to compute
> the L-R signal and the rds,
> just by subtracting delay times the phase difference.
> On line 791 for whatever reason the phase was computed based on the double
> delay
> 
> No idea where the "2 *" came from, it is removed now (and the delay not
> halved
> 
> best
> jan
> 
> 
> Op za 24 dec. 2022 om 01:36 schreef tomneda ***@***.***>:
> 
>> Hello Jan,
>> thanks that you like the mod.
>>
>> Yes you are right with the delay if you count only the loaded samples into
>> the pilot FFT filter, but the filtering itself takes the half time of the
>> underlying impulse response of the Sinc function? So I guess the
>> substracted filtersize has to be halfed. The additional minus 1 I got by
>> experiment and has a big effect if it is there or not.
>>
>> I already saw the new steep IF filtering. Your DX friend needs that really
>> for Wideband-FM or tries he (my experimental intended) AM mode where the 15
>> or 9 kHz are sensful and should be deeper?
>>
>> Maybe you make the filtersize dependend on the needed bandwidth, e.g.
>> using a big filtersize <20kHz and a smaller one above?
>>
>> Thanks for the x-max greetings I can only repeat this again.
>>
>> PS: meanwhile I sent you a further pull request, hope they are ok for you.
>>
>> BR
>> Thomas
>>
>> 23.12.2022 20:43:06 JvanKatwijk ***@***.***>:
>>
>> > Interesting, but strange
>> >
>> > The mod seems to work well, I'll test tomorrow with some other devices.
>> > The strange thing is that the delay of the pilotBandfilter is the size of
>> > the fft - the size of the fir filtersize that is used,
>> > the overlap size is here the same as the filtersize, so before spitting
>> out
>> > a filtered sample, a buffer with the size (fftsize - filtersize_
>> > is filled.
>> >
>> > One more comment:
>> > A radio friend/colegue was not very impressed by the band filtering, he
>> is
>> > a DX-er. So I modified the fit band filter with
>> > an FFT filter with a huge size, filtering is of course much better, but
>> cpu
>> > use increases when the filter is used, so
>> > I changed it such that if the filter id "OFF" it is not in the processing
>> > chain
>> >
>> >
>> > Anyway,
>> > I wish you fine Xmas and a happy - and healthy - 2023
>> >
>> > best
>> > jan
>> >
>> >
>> > Op vr 23 dec. 2022 om 18:12 schreef tomneda ***@***.***>:
>> >
>> >> Hi Jan,
>> >> as mentioned in private emails to you longer ago I still had a stereo
>> >> issue with my new SDRPlay RSPdx device that it sounds only like "mono"
>> >> whereas the DabStick works quite well.
>> >>
>> >> I go on checking this further and I have this pull request for you where
>> >> it at least these two devices work well, but the fix sounds reasonable
>> so I
>> >> expect that other devices will working well, too.
>> >>
>> >> The main change I did was the pilotDelay where the calculation is now
>> >> regarding the pilot filter pulse center not the whole filter length at
>> all:
>> >>
>> >> - new: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE / 2 - 1) *
>> >> OMEGA_PILOT;
>> >> - old: pilotDelay = (FFT_SIZE - PILOTFILTER_SIZE) * OMEGA_PILOT;
>> >>
>> >> What is not so clear to me, I had also change the mixing phase about 90
>> >> degree (at the doubled 38 kHz side) and so I used the sin() instead of
>> the
>> >> cos() function to do the LRDiff mix.
>> >>
>> >> What could be still a (new) issue: I am not sure if I should shift about
>> >> +90 or -90 degree. So it could be a minus in front of the sin necessary
>> (or
>> >> better/faster, shift +/-45 degree (we double later!) further in
>> pilotDelay.
>> >> With this issue we would have a swapped L/R channel. Maybe you have a
>> >> reliable test signal for this?
>> >> ------------------------------
>> >>
>> >> If you want to test my fix with other devices you will find a test
>> branch
>> >> in my repo:
>> >> https://github.com/tomneda/sdr-j-fm/tree/develop/stereo_issue_test
>> >>
>> >> Doing this for a test:
>> >>
>> >> 1. Select "S|S" to hear only the sideband signal.
>> >> 2. Check different stereo broadcast stations as some could have pilot
>> >> frequency inaccuracy.
>> >> 3. Check each station with the IQ balance slider whether the signal
>> >> get minimal at center "0" position
>> >>
>> >> The IQ balance slider adds an artificial phase shift (in degree) to the
>> >> pilot tone
>> >> ------------------------------
>> >>
>> >> BTW: I wish you this way a happy X-mas and so many health and luck what
>> >> you can get :-)
>> >>
>> >> BR
>> >> Thomas
>> >> ------------------------------
>> >> You can view, comment on, or merge this pull request online at:
>> >>
>> >> https://github.com/JvanKatwijk/sdr-j-fm/pull/10
>> >> Commit Summary
>> >>
>> >> - fee74a2
>> >> <
>> https://github.com/JvanKatwijk/sdr-j-fm/pull/10/commits/fee74a2f540900ddb292a1e36c2d97b27d426eda
>> >
>> >> Fix stereo issue with different SDR devices
>> >>
>> >> File Changes
>> >>
>> >> (2 files <https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files>)
>> >>
>> >> - *M* .gitignore
>> >> <
>> https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947
>> >
>> >> (1)
>> >> - *M* src/fm/fm-processor.cpp
>> >> <
>> https://github.com/JvanKatwijk/sdr-j-fm/pull/10/files#diff-e8d084e4e77aa8b8e699b0d278fedac65a272893601b67dfa5caf0d806cfb521
>> >
>> >> (4)
>> >>
>> >> Patch Links:
>> >>
>> >> - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.patch
>> >> - https://github.com/JvanKatwijk/sdr-j-fm/pull/10.diff
>> >>
>> >> —
>> >> Reply to this email directly, view it on GitHub
>> >> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10>, or unsubscribe
>> >> <
>> https://github.com/notifications/unsubscribe-auth/ACCPHQDM5DSGI2HLAGSS5KTWOXMOZANCNFSM6AAAAAATH5TMZ4
>> >
>> >> .
>> >> You are receiving this because you are subscribed to this thread.Message
>> >> ID: ***@***.***>
>> >>
>> >
>> >
>> > --
>> > Jan van Katwijk
>> >
>> > —
>> > Reply to this email directly, view it on GitHub[
>> https://github.com/JvanKatwijk/sdr-j-fm/pull/10#issuecomment-1364271809],
>> or unsubscribe[
>> https://github.com/notifications/unsubscribe-auth/AYYTND5CX3O3KT6BCUZE7IDWOX6ERANCNFSM6AAAAAATH5TMZ4
>> ].
>> > You are receiving this because you authored the thread.[Verfolgungsbild][
>> https://github.com/notifications/beacon/AYYTND2CFV2EQ3VNSCFLMJ3WOX6ERA5CNFSM6AAAAAATH5TMZ6WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSRKERMC.gif
>> ]
>> >
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/sdr-j-fm/pull/10#issuecomment-1364422930>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQED7WEEETO5WQT3QA3WOZAPNANCNFSM6AAAAAATH5TMZ4>
>> .
>> You are receiving this because you modified the open/close state.Message
>> ID: ***@***.***>
>>
> 
> 
> -- 
> Jan van Katwijk
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/JvanKatwijk/sdr-j-fm/pull/10#issuecomment-1364495176], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AYYTND7ABIEXAJS7S5BKFYTWO25HBANCNFSM6AAAAAATH5TMZ4].
> You are receiving this because you authored the thread.[Verfolgungsbild][https://github.com/notifications/beacon/AYYTND27R2YQR53MON2HO43WO25HBA5CNFSM6AAAAAATH5TMZ6WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSRKSFUQ.gif]
>
